py/plugins/google-genai/src/genkit/plugins/google_genai/google.py

```diff
@@ -206,6 +206,33 @@ def _resolve_embedder(self, ai: GenkitRegistry, name: str) -> None:
             fn=embedder.generate,
         )
 
+    def list_actions(self) -> list[dict[str, str]]:
+        """Generate a list of available actions or models.
+
+        Returns:
+            list of actions dicts with the following shape:
+            {
+                'name': str,
+                'kind': ActionKind,
+            }
+        """
+        actions_list = list()
+        for m in self._client.models.list():
+            name = m.name.replace('models/', '')
+            if 'generateContent' in m.supported_actions:
+                actions_list.append({
```

Comment:
we talked about `modelActionMetadata` and `embedderActionMetadata` helpers:
https://github.com/firebase/genkit/blob/ae1549c627c02dfb4ab9a7d011b1f3fcaeb76b90/js/plugins/googleai/src/index.ts#L235
this metadata must contain input and output schema, as well as all the necessary model/embedder info.